File: codegen/results/v002/gemma4_31b_think/20260414_083459/typescript/004_json_filter/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Interface defining the structure of the input records
 */
interface Record {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

function solve() {
  try {
    // Define the path to the input file
    const filePath = path.join(process.cwd(), 'input', 'data.json');

    // Read the file content
    const rawData = fs.readFileSync(filePath, 'utf8');

    // Parse the JSON data
    const data: Record[] = JSON.parse(rawData);

    // Filter: records where active is true and age is 30 or older
    const filteredData = data.filter((item) => item.active === true && item.age >= 30);

    // Sort: by name ascending
    const sortedData = filteredData.sort((a, b) => {
      if (a.name < b.name) return -1;
      if (a.name > b.name) return 1;
      return 0;
    });

    // Output the resulting JSON array to stdout
    process.stdout.write(JSON.stringify(sortedData, null, 2) + '\n');
  } catch (error) {
    // In a production environment, we'd handle specific errors (e.g., FileNotFound),
    // but based on the requirements, we just need the logic for the provided input.
    process.exit(1);
  }
}

// Execute the solver
solve();

File: codegen/results/v002/gemma4_31b_think/20260414_083459/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
)

// Record represents the structure of the input JSON objects.
type Record struct {
	Name   string  `json:"name"`
	Age    int     `json:"age"`
	Active bool    `json:"active"`
	Score  float64 `json:"score"`
}

func main() {
	// Read the file input/data.json
	filePath := "input/data.json"
	data, err := os.ReadFile(filePath)
	if err != nil {
		os.Exit(1)
	}

	// Parse the JSON array
	var records []Record
	err = json.Unmarshal(data, &records)
	if err != nil {
		os.Exit(1)
	}

	// Filter records where active is true and age is 30 or older
	var filtered []Record
	for _, r := range records {
		if r.Active && r.Age >= 30 {
			filtered = append(filtered, r)
		}
	}

	// Sort the filtered records by name ascending
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// Marshal the result back to JSON
	output, err := json.MarshalIndent(filtered, "", "  ")
	if err != nil {
		os.Exit(1)
	}

	// Output to stdout
	fmt.Println(string(output))
}